sr_fingertip_visualization/src/sr_fingertip_visualization/finger_widgets_graphs.py

```diff
@@ -0,0 +1,271 @@
+
+    def timerEvent(self):
+        for data_field in self._data_fields:
+            if self._data_checkboxes[data_field].isChecked():
+                self._plot.update_plot(self._data)
+
+
+class FingerWidgetGraphBiotac(FingerWidgetGraphGeneric):
+
+    def __init__(self, side, finger, parent):
+        super().__init__(finger, side, parent=parent)
+
+        self._data_fields = ['pac0', 'pac1', 'pdc', 'tac', 'tdc']
+        self._initialize_data_structure()
+        self._tactile_data_callback(rospy.wait_for_message('/{}/tactile'.format(self._side), BiotacAll))
+
+        self._plot_colors = list(self.plot_descriptors.keys())[:len(list(self._data.keys()))]
+        self._plot = GenericDataPlot(self._data, list(self.plot_descriptors.keys()))
+
+        plot_checkboxes = QGroupBox("Data")
+        plot_checkboxes.setSizePolicy(1, 2)
+        plot_checkboxes_layout = QGridLayout()
+        self._data_checkboxes = dict()
+        for i, data_field in enumerate(self._data_fields):
+            self._data_checkboxes[data_field] = QCheckBox(data_field)
+            self._data_checkboxes[data_field].setIcon(self.plot_descriptors[self._plot_colors[i]])
+
+        plot_checkboxes_layout.addWidget(self._data_checkboxes['pac0'], 0, 0, alignment=Qt.AlignLeft)
+        plot_checkboxes_layout.addWidget(self._data_checkboxes['pac1'], 0, 1, alignment=Qt.AlignLeft)
+        plot_checkboxes_layout.addWidget(self._data_checkboxes['pdc'], 0, 2, alignment=Qt.AlignLeft)
+        plot_checkboxes_layout.addWidget(self._data_checkboxes['tac'], 1, 0, alignment=Qt.AlignLeft)
+        plot_checkboxes_layout.addWidget(self._data_checkboxes['tdc'], 1, 1, alignment=Qt.AlignLeft)
+
+        plot_checkboxes.setLayout(plot_checkboxes_layout)
+
+        self._data_checkboxes['pac0'].clicked.connect(self.action_checkbox_pac0)
+        self._data_checkboxes['pac1'].clicked.connect(self.action_checkbox_pac1)
+        self._data_checkboxes['pdc'].clicked.connect(self.action_checkbox_pdc)
+        self._data_checkboxes['tac'].clicked.connect(self.action_checkbox_tac)
+        self._data_checkboxes['tdc'].clicked.connect(self.action_checkbox_tdc)
+
+        layout = QVBoxLayout()
+        layout.addWidget(plot_checkboxes, alignment=Qt.AlignTop)
+        layout.addWidget(self._plot)
+
+        self.setLayout(layout)
+        self.start_timer_and_subscriber()
+
+    def _initialize_data_structure(self):
+        self._data_fields = ['pac0', 'pac1', 'pdc', 'tac', 'tdc']
+        for data_field in self._data_fields:
+            self._data[data_field] = list()
+
+    def action_checkbox_pac0(self, state):
+        self._plot.show_trace('pac0', state)
+
+    def action_checkbox_pac1(self, state):
+        self._plot.show_trace('pac1', state)
+
+    def action_checkbox_pdc(self, state):
+        self._plot.show_trace('pdc', state)
+
+    def action_checkbox_tac(self, state):
+        self._plot.show_trace('tac', state)
+
+    def action_checkbox_tdc(self, state):
```

Comment:
You could avoid having this duplicate of code by extracting the name of the object that sent the signal (provided that you have named each object with the data_field)

Reply:
It's more a suggestion than anything else

Reply:
Really good point, appreciate that